jobs/build/ocp4/Jenkinsfile

```diff
@@ -148,15 +148,38 @@ node {
                 }
 
                 stage("update dist-git") { build.stageUpdateDistgit() }
-                stage("build images") { build.stageBuildImages() }
+                stage("build images") {
+                    build.stageBuildImages()
+                    branch=build.version.branch
+                    imagelist=build.buildPlan.imagesIncluded.split(',')
+                    for (i = 0; i < imagelist.size(); i++) {
+                        out = commonlib.shell(
+                            returnStdout: true,
+                            script: "brew list-tagged --latest-n=2 --quiet ${branch} | grep ${imagelist[i]} | grep .p0 | wc -l"
```

Comment:
a lot of problems in this one line.

in general, i do not like to see bash doing this much work unless it is in a bash script. Here you have groovy to do the work for you, and it has much more precise tools for processing output. so does python if you prefer.

note that this invokes brew to list _all_ components in the tag, once for every image (and then again below). this is not a quick command. it should be called _once_, and processed into a lookup hash, then the loop can iterate against that hash. the hash might look like:

```
  [
    "foo-container": ["v4.8.0-202104270102.p0"],  // there might only be one ever built
    "bar-container": ["v4.8.0-202104270102.p0", "v4.8.0-202104270304.p1"], // here's one that just went private
    "baz-container": ["v4.8.0-202104270102.p1", "v4.8.0-202104270304.p0"], // here's one that just went public
  ...]
```
(note all three of these would have `wc -l` of 1, but only one is what we're looking for.)
showing groovy notation of course - but invoking a small python script could be good too (then you could also choose to make koji api calls rather than invoke brew).

finally, `grep ${imagelist[i]}` could be a very inaccurate match for two reasons. first, the distgit name does not necessarily match the brew component at all. second, it could match more than one (for example `foo` and `foo-operator`).

instead, invoke doozer to get a list of the components you are looking for:
`doozer -i ${imagesIncluded} images:print --short '{component}'`
then you can lookup each in the hash built above.

Reply:
I got it, I will split this part in another PR and think of a better way to collect data.